src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -67,20 +73,21 @@ function getIDFromKey(key) {
 }
 
 function get(id) {
-  if (canUseMap) {
+  if (canUseCollections) {
     return itemMap.get(id);
+  } else {
+    var key = getKeyFromID(id);
+    return itemByKey[key];
   }
-  var key = getKeyFromID(id);
-  return itemByKey[key];
 }
 
 function remove(id) {
-  if (canUseMap) {
+  if (canUseCollections) {
     itemMap.delete(id);
-    return;
+  } else {
```

Comment:
I replaced early returns with branches because it makes these two paths clearer in my opinion.